app/Modules/SessionRequest/SessionRequestService.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace App\Modules\SessionRequest;
+
+use App\Modules\Bookings\Booking;
+use App\Modules\Bookings\Services\BookingsService;
+use App\Modules\Position\Position;
+use App\Modules\SessionRequest\Exceptions\SessionRequestAlreadyAcceptedException;
+use App\Modules\SessionRequest\Exceptions\SessionRequestAlreadyExistsException;
+use App\Modules\SessionRequest\Exceptions\SessionRequestNotAcceptedException;
+use App\User;
+use Carbon\Carbon;
+
+class SessionRequestService
+{
+    private BookingsService $bookingsService;
+
+    public function __construct(BookingsService $bookingsService)
+    {
+        $this->bookingsService = $bookingsService;
+    }
+
+    public function createSessionRequest(User $user, Position $position)
+    {
+        $existingSessionRequests = SessionRequest::where('user_id', $user->id)->where(
+            'position_id',
+            $position->id
+        )->get();
+
+        throw_if(! $existingSessionRequests->isEmpty(), SessionRequestAlreadyExistsException::class);
+
+        return SessionRequest::create([
+            'user_id' => $user->id,
+            'position_id' => $position->id,
+        ]);
+    }
+
+    public function revokeSessionRequest(SessionRequest $sessionRequest)
+    {
+        throw_if($sessionRequest->isTaken(), SessionRequestAlreadyAcceptedException::class);
```

Comment:
Might be good to be consistent with naming here - do we want to call everything 'accepted' or 'taken'? I would argue accepted is better than taken here

Reply:
yeah good point actually, I've done this at the database level too https://github.com/VATSIM-UK/ukts/pull/67/commits/dd3848a8dac67311fec30c932a178a4bf75ff472